crates/sui-sdk/src/transaction_builder.rs

```diff
@@ -100,6 +102,31 @@ impl TransactionBuilder {
         ))
     }
 
+    pub async fn pay(
+        &self,
+        signer: SuiAddress,
+        input_coins: Vec<ObjectID>,
+        recipients: Vec<SuiAddress>,
+        amounts: Vec<u64>,
+        gas: Option<ObjectID>,
+        gas_budget: u64,
+    ) -> anyhow::Result<TransactionData> {
+        let gas = self
+            .select_gas(signer, gas, gas_budget, input_coins.clone())
+            .await?;
+        let handles: Vec<_> = input_coins
```

Comment:
I think you can avoid cloning `input_coins` at 115 by moving the `handles` computation earlier.